File: src/main/java/dev/retrotv/crypto/owe/kdf/KDF.java
package dev.retrotv.crypto.owe.kdf;

import org.springframework.security.crypto.password.PasswordEncoder;

/**
 * 패스워드 암호화 클래스 구현을 위한 추상 클래스입니다.
 * {@link PasswordEncoder}를 상속받습니다.
 *
 * @author  yjj8353
 * @since   1.8
 */
public abstract class KDF implements PasswordEncoder {

    @Override
    public boolean matches(CharSequence rawPassword, String encodedPassword) {
        if (rawPassword == null || encodedPassword == null) {
            return false;
        }

        return encodedPassword.equals(encode(rawPassword));
    }
}

File: src/main/kotlin/dev/retrotv/crypto/owe/kdf/KDF.kt
package dev.retrotv.crypto.owe.kdf

import org.apache.logging.log4j.LogManager
import org.apache.logging.log4j.Logger
import org.springframework.security.crypto.password.PasswordEncoder

/**
 * 패스워드 암호화 클래스 구현을 위한 추상 클래스입니다.
 * [PasswordEncoder] 인터페이스를 상속받습니다.
 *
 * @author  yjj8353
 * @since   1.0.0
 */
abstract class KDF : PasswordEncoder {
    protected val log: Logger = LogManager.getLogger(this.javaClass)
}
